ludwig/schema/prompt.py

```diff
@@ -0,0 +1,74 @@
+from ludwig.api_annotations import DeveloperAPI
+from ludwig.schema import utils as schema_utils
+from ludwig.schema.utils import ludwig_dataclass
+
+
+@DeveloperAPI
+@ludwig_dataclass
+class RetrievalConfig(schema_utils.BaseMarshmallowConfig):
+    """This Dataclass is a schema for the nested retrieval config under prompt."""
+
+    type: str = schema_utils.String(
+        default=None,
+        allow_none=True,
+        description="The type of retrieval to use for the prompt.",
+    )
```

Comment:
Does it auto default to `random`? Or should we set a default type by default?

Reply:
It defaults to None, which means no samples are retrieved and it becomes zero-shot learning. I can make this description more clear on that front